[CI] CliApiKeyIT testCliConnectionWithApiKey failing

**Build Scans:**
- [elasticsearch-intake #35065 / part4](https://gradle-enterprise.elastic.co/s/wr4arbjdkoqie)
- [elasticsearch-pull-request #123269 / part-4](https://gradle-enterprise.elastic.co/s/rirzfwbpwmwru)
- [elasticsearch-intake #34967 / part4](https://gradle-enterprise.elastic.co/s/refojfe4qr6aw)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:sql:qa:server:security:without-ssl:javaRestTest" --tests "org.elasticsearch.xpack.sql.qa.security.CliApiKeyIT.testCliConnectionWithApiKey" -Dtests.seed=3888878B1218F492 -Dtests.locale=es-UY -Dtests.timezone=NZ -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.sql.qa.security.CliApiKeyIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testCliConnectionWithApiKey'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: unconsumed lines
Expected: an empty collection
     but: <[quit]>
```

**Issue Reasons:**
- [main] 3 failures in test testCliConnectionWithApiKey (0.4% fail rate in 701 executions)
- [main] 2 failures in step part4 (3.3% fail rate in 60 executions)
- [main] 2 failures in pipeline elasticsearch-intake (3.3% fail rate in 60 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -372,9 +372,6 @@ tests:
 - class: org.elasticsearch.reindex.management.ReindexManagementClientYamlTestSuiteIT
   method: test {yaml=reindex/30_cancel_reindex/Cancel running reindex returns response and GET confirms completed}
   issue: https://github.com/elastic/elasticsearch/issues/142079
-- class: org.elasticsearch.xpack.sql.qa.security.CliApiKeyIT
-  method: testCliConnectionWithApiKey
-  issue: https://github.com/elastic/elasticsearch/issues/143125
 - class: org.elasticsearch.packaging.test.DebMetadataTests
   method: test05CheckLintian
   issue: https://github.com/elastic/elasticsearch/issues/142819

--- a/x-pack/plugin/sql/qa/server/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/CliApiKeyIT.java
+++ b/x-pack/plugin/sql/qa/server/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/CliApiKeyIT.java
@@ -12,6 +12,7 @@ import org.elasticsearch.xpack.sql.qa.cli.EmbeddedCli.ApiKeySecurityConfig;
 
 import static org.elasticsearch.xpack.sql.qa.security.RestSqlIT.SSL_ENABLED;
 import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertEquals;
 
 /**
  * Integration tests for CLI connections using API key authentication.
@@ -65,6 +66,7 @@ public class CliApiKeyIT extends SqlApiKeyTestCase {
             cli.readLine(); // separator line
             String valueLine = cli.readLine();
             assertThat(valueLine, containsString("123"));
+            assertEquals("", cli.readLine());
         }
     }
 

